### PR TITLE
sing up / login integration

### PR DESCRIPTION
**Pull request by kwlodarczyk0**

### CODE REVIEW

File: src/app/context/UserContext.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    update: setUser,
+    logout,
+  }
+
+  return <UserContext.Provider value={value}>{children}</UserContext.Provider>
+}
+
+export function useUser() {
+  return useContext(UserContext)
+}
```
Comment (tdsnxsuplewski-mateusz) [inline src/app/context/UserContext.tsx:49]: Do we actually need that user context?
We are already checking if jwtToken exists in the GuardedRoute( localstorage.getItem ) which indicates if user is logged or not

File: src/components/Nav/Nav.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      width: 100px;
+    }
+  }
+
+  &__user {
+    display: flex;
+    align-items: center;
+    gap: 20px;
+  }
+}
```
Comment (tdsnxsuplewski-mateusz) [inline src/components/Nav/Nav.scss]: I would prefer to move it to the styled.ts file, this way we will not move between styled components and pure css classes

File: src/app/helpers/GuardedRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    return <Element {...rest} />
+  } else {
+    if (!localStorage.getItem('jwtToken')) {
+      return <Navigate to="/login" replace />
+    } else {
+      return <Element {...rest} />
+    }
+  }
+}
+export default GuardedRoute
```
Comment (tdsnxsuplewski-mateusz) [inline src/app/helpers/GuardedRoute.tsx:11]: body of component can be simplified to: ( after we remove userContext )
```javascript
    if (!localStorage.getItem('jwtToken')) return <Navigate to="/login" replace />
      return <Element {...rest} />

```

**Review by tdsnxsuplewski-mateusz — CHANGES_REQUESTED** [review]
Check comments for details

File: src/app/context/UserContext.tsx (hunk 1 above)
Comment (kwlodarczyk0) [reply src/app/context/UserContext.tsx:49]: now i remove this functionality but probably we will have to use context later (it depends how our api will be implemented)

**Review by tdsnxsuplewski-mateusz — APPROVED** [review]
Ready to be merged

**Pull request merged**